specs/eventing/data-plane.md

```diff
@@ -1,125 +1,179 @@
+which receive an error when probing with HTTP OPTIONS SHOULD proceed using the
+HTTP POST mechanism.
+
+### Event Acknowledgement and Delivery Retry
+
+Event recipients MUST use the HTTP response code to indicate acceptance of an
+event. The recipient SHOULD NOT return a response accepting the event until it
+has handled event (processed the event or stored it in stable storage). The
+following response codes are explicitly defined; event recipients MAY also
+respond with other response codes. A response code not in this table SHOULD be
+treated as a retriable error.
+
+| Response code | Meaning                           | Retry | Delivery completed | Error |
+| ------------- | --------------------------------- | ----- | ------------------ | ----- |
+| `1xx`         | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `200`         | [Event reply](#event-reply)       | No    | Yes                | No    |
+| `202`         | Event accepted                    | No    | Yes                | No    |
+| other `2xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| other `3xx`   | (Unspecified)                     | No\*  | No\*               | Yes\* |
+| `400`         | Unparsable event                  | No    | No                 | Yes   |
+| `404`         | Endpoint does not exist           | Yes   | No                 | Yes   |
+| `409`         | Conflict / Processing in progress | Yes   | No                 | Yes   |
+| `429`         | Too Many Requests / Overloaded    | Yes   | No                 | Yes   |
+| other `4xx`   | Error                             | No    | No                 | Yes   |
+| other `5xx`   | Error                             | Yes   | No                 | Yes   |
+
+\* Unspecified `1xx`, `2xx`, and `3xx` response codes are **reserved for future
+extension**. Event recipients SHOULD NOT send these response codes in this spec
+version, but event senders MUST handle these response codes as errors and
+implement appropriate failure behavior.
+
+Recipients MUST be able to handle duplicate delivery of events (for example, via
+idempotency or tracking event delivery state) and MUST accept delivery of
+duplicate events, as the event acknowledgement could have been lost in return to
+the sender. As specified in the
+[CloudEvents specification](https://github.com/cloudevents/spec/blob/v1.0.1/primer.md#id),
+event recipients MUST use the
+[`source` and `id` attributes](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#required-attributes)
+to determine duplicate events if needed. This specification does not describe
+state requirements for clients which need to detect duplicate events. (see
+[observability](#observability) for an example case where other event attributes
+may vary from one delivery attempt to another).
+
+Where possible, event senders SHOULD re-attempt delivery of events where the
```

Comment:
I think SHOULD may be too strong here. In serving we specifically stopped the networking layer from retrying on 5xx errors because it ended up spamming the ksvc when the ksvc was correctly returning a 503.  Even with a MAY I'd still like to see some text that talks about how the sender needs to be selective about it to avoid DDOSing the sink.

Reply:
That's a good thing to mention, but it still pushes people towards resending... just slower. I keep having flashbacks to wasting my time debugging a ksvc trying to figure out why I was seeing 3 times the errors in its logs... and it was because of the network retries.  I think it's sufficient to talk about how it's ok to retry if the sender thinks a new result will happen, but not just because it failed with a 5xx